runtime/jvmti/jvmtiHelpers.c

```diff
@@ -1419,11 +1464,24 @@ setEventNotificationMode(J9JVMTIEnv * j9env, J9VMThread * currentThread, jint mo
 	if (event_thread == NULL) {
 		eventMap = &(j9env->globalEventEnable);
 	} else {
+		j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(event_thread);
 		rc = getVMThread(currentThread, event_thread, &targetThread, TRUE, TRUE);
 		if (rc != JVMTI_ERROR_NONE) {
 			goto done;
 		}
-		eventMap = &(THREAD_DATA_FOR_VMTHREAD(j9env, targetThread)->threadEventEnable);
+#if JAVA_SPEC_VERSION >= 19
+		rc = allocateTLS(vm, threadObject);
+		if (JVMTI_ERROR_NONE != rc) {
+			releaseVMThread(currentThread, targetThread);
+			goto done;
+		}
+#endif /* JAVA_SPEC_VERSION >= 19 */
+		rc = createThreadData(j9env, targetThread, threadObject);
```

Comment:
Actually I think it makes more sense to keep the functions separate since they are unrelated: `allocateTLS` allocates the TLS array and `createThreadData` allocates elements of the array. `allocateTLS` is only used for jdk19+ so it is not necessarily a part of creating thread data.